### ISSUE TITLE
[BUG] DllNotFound (webrtc) on Android / IL2CPP / Armv64

### ISSUE THREAD

**Issue by streunerlein** [Unity-Technologies/com.unity.webrtc#513]
When calling WebRTC.Initialize() there is a DllNotFoundException thrown on device (Android with IL2CPP and only Armv64 selected):

```
Error Unity DllNotFoundException: Unable to load DLL 'webrtc'. Tried the load the following dynamic libraries: Unable to load dynamic library 'webrtc' because of 'Failed to open the requested dynamic library (0x06000000) dlerror() = dlopen failed: library "webrtc" not found
 at Unity.WebRTC.NativeMethods.RegisterDebugLog (Unity.WebRTC.DelegateDebugLog func) [0x00000] in <00000000000000000000000000000000>
 at Unity.WebRTC.WebRTC.Initialize (Unity.WebRTC.EncoderType type, System.Boolean limitTextureSize, System.Boolean forTest) [0x00000] in <00000000000000000000000000000000>
```

1. Add the webrtc package version 2.4.0-exp.4 to a new unity project
2. Build & Run on device
3. Watch logcat

**Environment (please complete the following information):**
 - OS: Android
 - Unity Version 2021.2.0b6
 - Package version 2.4.0-exp.4
 - IL2CPP Armv64

**Comment by karasusan** [comment]
We tested it on Unity LTS version 2020.3 and 2019.4, but not 2021.2.
Could you check it on 2020.3?

**Comment by karasusan** [comment]
I reproduced it.
This is an urgent issue. Thanks for your report.

**Comment by streunerlein** [comment]
Thank you. I suspect it is an isssue with 2021.2, I have made a similar experience with .jar files and even filed a bug for it: https://issuetracker.unity3d.com/issues/android-il2cpp-jar-file-not-included-in-build-when-building-apk-with-il2cpp-and-only-arm64-target-architecture-selected

Probably the two issues are related?

**Comment by streunerlein** [comment]
Also adding https://issuetracker.unity3d.com/issues/only-some-assemblies-fail-to-be-loaded-when-assembly-name-does-not-match-the-file-name here, as it could be related as well.

**Comment by karasusan** [comment]
@streunerlein 
Wow, this is a valuable info. Thank you very much.

**Comment by karasusan** [comment]
I checked the issue with Unity2021.1.16 but not reproduced.
As you said, the cause of the issue seems to be the Unity editor 2021.2.

**Comment by streunerlein** [comment]
@karasusan Is there something more I can do? Is this being investigated by Unity (via you) or should I file another bug report?

Also, if anyone knows a working workaround that would be amazing. We are crying here because we are so close to get rid of [MixedReality-WebRTC](https://github.com/microsoft/MixedReality-WebRTC) but can't because of this.

**Comment by karasusan** [comment]
@streunerlein 
I am asking about this issue my colleagues.

**Comment by streunerlein** [comment]
It seems as if https://issuetracker.unity3d.com/issues/android-il2cpp-jar-file-not-included-in-build-when-building-apk-with-il2cpp-and-only-arm64-target-architecture-selected is fixed for **2022.1** and planned for **2021.2**. As this is the version I am on, I would like to keep this issue open until 2021.2 is fixed as well.

**Comment by streunerlein** [comment]
Update: In 2021.20b**13** this is indeed fixed by Unity, we've just run a successful test. Thank you @karasusan
